bridgeservice/bridge.go

```diff
@@ -1050,6 +1174,115 @@ func (b *BridgeService) GetLastReorgEventHandler(c *gin.Context) {
 	c.JSON(http.StatusOK, reorgEvent)
 }
 
+// GetRemoveGEREventsHandler retrieves remove GER events.
+//
+// @Summary Get remove GER events
+// @Description Returns a list of remove GER events, optionally filtered by block range or specific GER
+// @Tags ger-events
+// @Param from_block query uint64 false "Start block number for filtering"
+// @Param to_block query uint64 false "End block number for filtering"
+// @Param global_exit_root query string false "Filter by specific Global Exit Root hash"
+// @Produce json
+// @Success 200 {object} types.RemoveGEREventsResult "List of remove GER events"
+// @Failure 400 {object} types.ErrorResponse "Bad Request"
+// @Failure 500 {object} types.ErrorResponse "Internal Server Error"
+// @Failure 503 {object} types.ErrorResponse "Service Unavailable"
+// @Router /remove-ger-events [get]
+func (b *BridgeService) GetRemoveGEREventsHandler(c *gin.Context) {
+	b.logger.Debugf("GetRemoveGEREvents request received")
+
+	statusCode := http.StatusOK
+	startTime := time.Now()
+	defer func() {
+		reportMetrics(metrics.GetRemoveGEREventsReq, statusCode, startTime)
+	}()
+
+	ctx, cancel := context.WithTimeout(c, b.readTimeout)
+	defer cancel()
+
+	// Check if L2GERSyncer is available
+	if b.injectedGERs == nil {
+		statusCode = http.StatusServiceUnavailable
+		c.JSON(statusCode, gin.H{"error": "L2 GER syncer is not available"})
+		return
+	}
+
+	// Parse query parameters
+	fromBlockStr := c.Query("from_block")
+	toBlockStr := c.Query("to_block")
+	globalExitRootStr := c.Query("global_exit_root")
+
+	// Parse and validate parameters
+	var globalExitRoot *common.Hash
+	var fromBlock, toBlock *uint64
+
+	if globalExitRootStr != "" {
+		if !isValidHexHash(globalExitRootStr) {
+			statusCode = http.StatusBadRequest
+			c.JSON(statusCode, gin.H{
+				"error": "invalid global_exit_root parameter, must be a valid 32-byte hex hash (66 characters including 0x prefix)",
+			})
+			return
+		}
+		ger := common.HexToHash(globalExitRootStr)
+		globalExitRoot = &ger
+	}
+
+	if fromBlockStr != "" {
+		parsed, err := strconv.ParseUint(fromBlockStr, 10, 64)
+		if err != nil {
+			statusCode = http.StatusBadRequest
+			c.JSON(statusCode, gin.H{"error": "invalid from_block parameter"})
+			return
+		}
+		fromBlock = &parsed
+	}
+
+	if toBlockStr != "" {
+		parsed, err := strconv.ParseUint(toBlockStr, 10, 64)
+		if err != nil {
+			statusCode = http.StatusBadRequest
+			c.JSON(statusCode, gin.H{"error": "invalid to_block parameter"})
+			return
+		}
+		toBlock = &parsed
+	}
+
+	if fromBlock != nil && toBlock != nil && *fromBlock > *toBlock {
+		statusCode = http.StatusBadRequest
+		c.JSON(statusCode, gin.H{"error": "from_block must be less than or equal to to_block"})
+		return
+	}
+
+	// Get filtered remove events using single consolidated function
+	removeEvents, err := b.injectedGERs.GetRemoveGEREvents(ctx, globalExitRoot, fromBlock, toBlock)
```

Comment:
maybe we can send latest 50 events or something? not all
because earlier remove ger are not needed now